src/shogun/base/AnyParameter.h

```diff
@@ -135,6 +139,20 @@ namespace shogun
 		{
 			m_attribute_mask &= ~other;
 		}
+		std::string to_string() const
+		{
+			std::stringstream ss;
+			ss << "Description: "<< m_description << " with attributes: [";
+			for (auto it = begin<ParameterProperties>(); it != end<ParameterProperties>(); ++it)
+			{
+				if (it != begin<ParameterProperties>())
+					ss << ", ";
+				if (has_property(*it))
+					ss << static_cast<std::underlying_type_t<ParameterProperties>>(*it);
```

Comment:
Should do casting with `underlying_type` in the other methods, i.e. `EGradientAvailability get_gradient() const`

Reply:
Also might be worth printing the bit position rather than the int representation no?

Reply:
i would rather like to have a string representation of the flags :)

Reply:
in that case shouldn't you keep a map with a readable string representation, i.e. {ParameterProperties::HYPER: "hyperparameter"} and print the values?

Reply:
because right now this is just writing the integers right?